fapolicy_analyzer/ui/ancillary_trust_database_admin.py

```diff
@@ -130,11 +155,45 @@ def on_files_added(self, files):
 
     def on_trustBtn_clicked(self, *args):
         if self.selectedFiles:
-            self.add_trusted_files(*self.selectedFiles)
+            dne_list = [f for f in self.selectedFiles if not os.path.isfile(f)]
+            selectedFiles = [f for f in self.selectedFiles if f not in dne_list]
+            if len(dne_list) > 0:
+                removeDialog = RemoveDeletedDialog(deleted=dne_list).get_ref()
+                resp = removeDialog.run()
+                removeDialog.destroy()
+                if resp == Gtk.ResponseType.APPLY:
+                    self.delete_trusted_files(*dne_list)
+                    treeView = self.trustFileList.get_object("treeView")
+                    treeView.get_selection().select_path(Gtk.TreePath.new_first())
+                    model, pathlist = treeView.get_selection().get_selected_rows()
+                    if model:
+                        child_model = model.get_model().get_model()
+                        iter_ = child_model.get_iter(next(iter(pathlist)))  # single select use first pat
+                        child_model.remove(iter_)
+
+            if selectedFiles:
+                self.add_trusted_files(*selectedFiles)
 
     def on_untrustBtn_clicked(self, *args):
         if self.selectedFiles:
-            self.delete_trusted_files(*self.selectedFiles)
+            dne_list = [f for f in self.selectedFiles if not os.path.isfile(f)]
+            selectedFiles = [f for f in self.selectedFiles if f not in dne_list]
+            if len(dne_list) > 0:
+                removeDialog = RemoveDeletedDialog(deleted=dne_list).get_ref()
+                resp = removeDialog.run()
+                removeDialog.destroy()
+                if resp == Gtk.ResponseType.APPLY:
+                    self.delete_trusted_files(*dne_list)
+                    treeView = self.trustFileList.get_object("treeView")
+                    treeView.get_selection().select_path(Gtk.TreePath.new_first())
+                    model, pathlist = treeView.get_selection().get_selected_rows()
+                    if model:
+                        child_model = model.get_model().get_model()
+                        iter_ = child_model.get_iter(next(iter(pathlist)))  # single select use first pat
+                        child_model.remove(iter_)
```

Comment:
This code looks the same as what is in the `on_trustBtn_click` method. Can we abstract it to a helper method and DRY this up a bit?

Reply:
Good call. Got it condensed and cleaned up